### ISSUE TITLE
Stopping or restarting node.js debugger crashes vscode

### ISSUE THREAD

**Issue by aheitzmann** [Microsoft/vscode#14187]
- VSCode Version: 1.6.1
- OS Version: OSX El Capitan 10.11.6
- Node version: 6.7.0

Steps to Reproduce:
1. Start vscode through the finder
2. Open a new folder
3. Create a new file app.js in the root of the folder with contents

```
var msg = 'hello world'
console.log(msg)
```
1. [optional] Confirm that "hello world" is printed when running the file with
   `node app.js`
2. Click the gear in the DEBUG pane to create a Node.js launch.json config.
3. Hit F5 or click the run button to run app.js with the debugger attached
   Expected behavior: "hello world" is printed to the console. No breakpoints are set, so the program runs and terminates, and the debugger detaches automatically.
   Actual behavior: "hello world" is printed to the console. The debugger stays attached.
4. Hit Shift-F5 or the Stop Debugging button to detach the debugger and terminate the node program.
   Expected behavior: The debugger detaches and vscode exits debug mode.
   **Actual behavior: Nothing happens for about 10 seconds, then vscode crashes.**

_Possible clue_ 
If vscode was launched from the command line with 
`code`
(see the shell command setup here https://code.visualstudio.com/docs/setup/mac)
Then step 7 produces the following output in the debug console, and vscode does not crash.

```
/usr/local/Cellar/node/6.7.0/bin/node[57339]: ../src/debug-agent.cc:149:void node::debugger::Agent::Stop(): Assertion `(err) == (0)' failed.
 1: node::Abort() [/usr/local/bin/node]
 2: node::RunMicrotasks(v8::FunctionCallbackInfo<v8::Value> const&) [/usr/local/bin/node]
 3: node::debugger::Agent::~Agent() [/usr/local/bin/node]
 4: node::debugger::Agent::~Agent() [/usr/local/bin/node]
 5: node::Environment::~Environment() [/usr/local/bin/node]
 6: node::Start(int, char**) [/usr/local/bin/node]
 7: start [/usr/local/bin/node]
```

**Comment by roblourens** [comment]
Could you try with `"type": "node2"`, our experimental new debugger, to see if it works any better? I'm also curious whether you see this with different versions of Node, or in vscode insiders build - code.visualstudio.com/insiders.

**Comment by weinand** [comment]
- app.js not terminating properly because of this node bug: https://github.com/nodejs/node/issues/1788
- I cannot reproduce the crashing with VS Code 1.6.1, macOS 10.11.6, and node 6.7.0.

@aheitzmann It is possible that this issue is caused by an installed extension. Can you try to run VS Code without extensions? From the command line, execute: `code --disable-extensions` and try your steps again to see if it reproduces.

@roblourens node-debug2 shows the same behaviour: app.js does not terminate properly but crashing does not occur.

**Comment by roblourens** [comment]
That part is expected because of this bug unfortunately - https://github.com/Microsoft/vscode-node-debug2/issues/11

**Comment by aheitzmann** [comment]
@weinand  Unfortunately the crash does not reproduce when vscode is launched from the command line. I get the "Assertion `(err) == (0)' failed" error in that case whether or not --disable-extensions is specified.

I tried simply uninstalling all of my extensions, and the crash still reproduces when launched from the finder.

My user settings are
```
// Place your settings in this file to overwrite the default settings
{
    "editor.rulers": [100],
    "editor.wrappingIndent": "indent",
    "diffEditor.ignoreTrimWhitespace": false,
    "files.trimTrailingWhitespace": true,
    "files.autoSave": "afterDelay",
    "typescript.useCodeSnippetsOnMethodSuggest": true,

    "vim.useCtrlKeys": true
}
```
and I'm attaching a zip of the project in case that helps with the repro.
[nodecrashrepro.zip](https://github.com/Microsoft/vscode/files/545597/nodecrashrepro.zip)

**Comment by aheitzmann** [comment]
@roblourens The crash also reproduces with the `"type": "node2"` debugger.

**Comment by aheitzmann** [comment]
@roblourens The crash also reproduces with the latest vscode insiders build.

**Comment by lucas-natraj** [comment]
I've started seeing this issue as well with vscode 1.8.1, El Capitan, node version 6.9.1
Things were working just fine until yesterday, although I'm not sure what I have changed since then.
There was an update to vscode that I did install, but trying to revert to an earlier version still results in a  crash on debugger detach.

Has anyone else experiencing this? Since the last update to this issue was in October.
